sdcm/nemesis.py

```diff
@@ -3562,12 +3563,14 @@ def disrupt_add_remove_dc(self) -> None:
             InfoEvent(message='Running full cluster repair on each node').publish()
             for node in self.cluster.nodes:
                 node.run_nodetool(sub_cmd="repair -pr", publish_event=True)
-            self._write_read_data_to_multi_dc_keyspace(datacenters)
+            self._write_read_data_to_multi_dc_keyspace()
         self.cluster.decommission(new_node)
         self.monitoring_set.reconfigure_scylla_monitoring()
         datacenters = list(self.tester.db_cluster.get_nodetool_status().keys())
         assert not [dc for dc in datacenters if dc.endswith("_nemesis_dc")], "new datacenter was not unregistered"
         self._verify_multi_dc_keyspace_data(consistency_level="QUORUM")
+        with self.cluster.cql_connection_patient(node) as session:
+            session.execute('DROP KEYSPACE keyspace_new_dc')
```

Comment:
also maybe it should be done with a try/finally or a context manager, to make sure that it would happen even if there are errors before, during the decommission of any assertion that would pop, it would never get here.

Reply:
done